zcash_client_backend/src/welding_rig.rs

```diff
@@ -275,10 +275,10 @@ pub fn scan_block<P: consensus::Parameters, K: ScanningKey>(
         }
 
         if !(shielded_spends.is_empty() && shielded_outputs.is_empty()) {
-            let mut txid = TxId([0u8; 32]);
-            txid.0.copy_from_slice(&tx.hash);
+            let mut hash = [0u8; 32];
+            hash.copy_from_slice(&tx.hash);
```

Comment:
We can use `tx.hash.try_into()` here, but ideally this would just be a `tx.txid()` method on `CompactTx` (which we can impl in the `zcash_client_backend::proto` module).